src/driver/safe/alloc.rs

```diff
@@ -107,10 +107,25 @@ impl CudaDevice {
 }
 
 impl CudaDevice {
+    pub fn is_async(self: &Arc<Self>) -> Result<bool, result::DriverError> {
+        let value = unsafe {
+            result::device::get_attribute(
+                self.cu_device,
+                sys::CUdevice_attribute_enum::CU_DEVICE_ATTRIBUTE_MEMORY_POOLS_SUPPORTED,
+            )?
```

Comment:
Thoughts on storing this as an attribute of CudaDevice, so we only have to run it once when CudaDevice is created?

Reply:
I just wanted to keep it simple first, but quite easy to do.

Reply:
Done.